nameserver/server.go

```diff
@@ -173,15 +246,33 @@ func (s *DNSServer) queryHandler(lookups []Lookup) dns.HandlerFunc {
 	}
 }
 
-func (s *DNSServer) rdnsHandler(lookups []Lookup, client *dns.Client) dns.HandlerFunc {
-	fallback := s.notUsHandler(client)
+func (s *DNSServer) rdnsHandler(lookups []Lookup, proto dnsProtocol) dns.HandlerFunc {
+	fallback := s.notUsHandler(proto)
 	return func(w dns.ResponseWriter, r *dns.Msg) {
 		q := r.Question[0]
+		maxLen := getMaxReplyLen(r, proto)
 		Debug.Printf("Reverse query: %+v", q)
 		if q.Qtype == dns.TypePTR {
```

Comment:
If this type is not matched, we return nothing? Refused or Not Implemented might be better?